src/software/sensor_fusion/sensor_fusion_test.cpp

```diff
@@ -379,3 +407,41 @@ TEST_F(SensorFusionTest, test_referee_blue_then_normal)
     World result_2 = *sensor_fusion.getWorld();
     EXPECT_EQ(expected_2, result_2.gameState());
 }
+
+TEST_F(SensorFusionTest, ball_placement_friendly_set_by_referee)
+{
+    SensorProto sensor_msg;
+
+    // send Point(50, 75) to Referee message
+    *(sensor_msg.mutable_ssl_referee_msg()) = *referee_ball_placement_yellow;
+
+    auto ssl_wrapper_packet =
+        createSSLWrapperPacket(std::move(geom_data), initDetectionFrame());
+    // set vision msg so that world is valid
+    *(sensor_msg.mutable_ssl_vision_msg()) = *ssl_wrapper_packet;
+
+    sensor_fusion.updateWorld(sensor_msg);
+    World result = *sensor_fusion.getWorld();
+
+    Point returned_point = result.gameState().getBallPlacementPoint().value();
+    EXPECT_EQ(Point(50, 75), returned_point);
+}
+
+TEST_F(SensorFusionTest, ball_placement_enemy_set_by_referee)
+{
+    SensorProto sensor_msg;
+
+    // send Point(20, 35) to Referee message
+    *(sensor_msg.mutable_ssl_referee_msg()) = *referee_ball_placement_blue;
+
+    auto ssl_wrapper_packet =
+        createSSLWrapperPacket(std::move(geom_data), initDetectionFrame());
+    // set vision msg so that world is valid
+    *(sensor_msg.mutable_ssl_vision_msg()) = *ssl_wrapper_packet;
+
+    sensor_fusion.updateWorld(sensor_msg);
+    World result = *sensor_fusion.getWorld();
+
+    std::optional<Point> returned_point = result.gameState().getBallPlacementPoint();
+    EXPECT_EQ(std::nullopt, returned_point);
```

Comment:
can we add a quick comment on why we expect `std::nullopt` to be returned

Reply:
fixed